### PR TITLE
Add main workflow to lint code

### PR DESCRIPTION
**Pull request by andraskadar**
Add a main workflow file that runs on each update on the main branch and pull requests and checks for linting errors.

Future improvements:
* Run the tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          path: ~/.mint
+          key: ${{ runner.os }}-mint-${{ hashFiles('Mintfile') }}
+          restore-keys: ${{ runner.os }}-mint-
+
+      - name: Install Mint
+        run: brew install mint
+
+      - name: Install Mint Packages
+        if: steps.mint-cache-v2.outputs.cache-hit != 'true'
+        run: mint bootstrap
```
Comment (andraskadar) [inline .github/workflows/main.yml:30]: This is a bit "tricky" part. We have set up the linting & formatting to install dependencies via Mint. Now we need to install mint on CI to resolve the dependencies.

How Mint works, it downloads the dependencies and builds an executable Swift package out of the source codes. In order to not perform the building every time, we set up caching on the Mint directory and only call `Install Mint Packages` if the cache is not available.

The cache:
- `path`: which directory to cache
- `key`: the key that should indicate if the cache is valid or not. You can see it hashes the Mintfile where we are locking the versions of the dependencies. If no change in the Mintfile => we can re-use the latest dependencies
- `restore-keys`: basically a fall-back `key` when there is a mismatch in e.g. Mintfile to use caches that are matching at least the beginning of the key to call bootstrap from the latest good cache

The cache is automatically caching before finishing up the workflow.

We only use the `id` of the cache step to reference in the `Install Mint Packages` the result of the caching to be able to skip install if there were caches.
|Without cache hit|With cache hit|
|-|-|
|Screen Shot 2022-05-02 at 1 03 41 AM|Screen Shot 2022-05-02 at 1 03 12 AM|

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Run Swiftformat linter
+        run: mint run swiftformat . --lint
+
+      - name: Run Swiftlint
+        run: mint run swiftlint lint --reporter github-actions-logging
```
Comment (andraskadar) [inline .github/workflows/main.yml:36]: These are basic `bash` commands. The trick here is the `--reporter github-actions-logging` which tells swiftlint to report in a GH Actions format.

This basically means that the format changes from 
```
/Users/andraskadar/Development/furry/furry-fiesta-ios/Log/Tests macOS/Tests_macOS.swift:10:7: error: Type Name Violation: Type name should only contain alphanumeric characters: 'Tests_macOS' (type_name)
```
to
```
::error file=Log/Tests macOS/Tests_macOS.swift,line=10,col=7::Type name should only contain alphanumeric characters: 'Tests_macOS' (type_name)
```

and you can see in the changed files list how GH will automatically parse these errors and warnings to present in-line with the code.

Screen Shot 2022-05-02 at 1 08 04 AM

**Review by andraskadar — COMMENTED** [review]
Nem vartalak meg a pair programmingra, mert csak most este volt idom, de irtam commenteket a kodhoz magyarazatkent. Ha valami nem vilagos, irj.

Meg valoszinuleg kell egy follow-up Swiftlint javitas mielott mergelni lehet 😓

**Review by danieltmbr — APPROVED** [review]
Noice, thanks for the explanations.

**Pull request merged**